### DISCUSSION TITLE
[destination-tidb, normalization] how to change `varchar(1000)` limit?

### DISCUSSION THREAD

**Discussion by yevhen-korientsov** [airbytehq/airbyte#33710] [Connector Questions]
Hi,
we have a connection from Mongodb to Tidb, airbyte creates schema automatically, and it creates varchar(1000) data type in Tidb. 
And we have some fields which are longer than 1000, so we faced error:
1406 (22001): Data Too Long, field len 1000, data len 1101

Is there some simple way to increase this value from 1000 ? (Tidb has limit 16383)

**Comment by marcosmarxm** [comment]
Today the destination TiDB uses Airbyte normalization, a deprecated features, to create the final tables.
You can check [here](https://github.com/airbytehq/airbyte/blob/bfe9d8b466498d9417253b52812a6870e99959df/airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/datatypes.sql#L70-L72) where this happens. If you need to change that, you probably need to edit the normalization module and rebuild it from scratch to uses the new version.
Another way and more practical is to export the generated dbt project and use a custom transformation, instructions you can read here: https://docs.airbyte.com/operator-guides/transformation-and-normalization/transformations-with-sql

[a comment by an automated account is not shown]
